bindings/python/mainloop.pyx

```diff
@@ -0,0 +1,17 @@
+
+cdef extern from "sol-mainloop.h":
+	int sol_init()
+	int sol_run()
+	void sol_shutdown()
+
+class Mainloop():
+	def __init__(self):
+		sol_init()
+
+	def __del__(self):
+		sol_shutdown()
```

Comment:
I was leaning towards implementing asyncio's EventLoopPolicy interface and call into soletta mainloop functions.  I think this is your latter suggestion.  We'd implement "run_forever" like this:

```python

class SolettaEventLoopPolicy(asyncio.BaseEventLoopPolicy):
def __init__(self):
   self.mainloop = soletta.MainLoop

def run_forever(self):
    self.mainloop.run()

def call_later(self, ...):
   self.mainloop.add_timeout(...)